glide-core/redis-rs/redis/src/cluster_async/mod.rs

```diff
@@ -1518,16 +1519,24 @@ where
 
         let mut res = Ok(());
         if !skip_slots_refresh {
-            let retry_strategy = ExponentialBackoff {
-                initial_interval: DEFAULT_REFRESH_SLOTS_RETRY_INITIAL_INTERVAL,
-                max_interval: DEFAULT_REFRESH_SLOTS_RETRY_MAX_INTERVAL,
-                max_elapsed_time: None,
-                ..Default::default()
-            };
+            let retry_strategy = ExponentialFactorBackoff::from_millis(
+                DEFAULT_REFRESH_SLOTS_RETRY_BASE_DURATION_MILLIS,
+                DEFAULT_REFRESH_SLOTS_RETRY_BASE_FACTOR,
+            )
+            .map(jitter_range(0.8, 1.2))
+            .take(DEFAULT_NUMBER_OF_REFRESH_SLOTS_RETRIES);
             let retries_counter = AtomicUsize::new(0);
```

Comment:
Note the we don't use it as a limit anymore, only for the sake of choosing the topology.

Reply:
I see I tried to see if there's a way to call something different on last try. It appears that the only callback is notify, which can be used, but it's not the reason it exists, so probably not a good idea.
In general, I would prefer the called function to have one logic, and the caller to call functions base on needs, so we keep functions shorter and more readable. That's why I want to avoid the counter logic.
But couldn't find a replacement currently.